grakn-graql/src/main/java/ai/grakn/graql/internal/query/aggregate/StdAggregate.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Grakn - A Distributed Semantic Database
+ * Copyright (C) 2016  Grakn Labs Limited
+ *
+ * Grakn is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ *
+ */
+
+package ai.grakn.graql.internal.query.aggregate;
+
+import ai.grakn.concept.Concept;
+import ai.grakn.graql.VarName;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.OptionalDouble;
+import java.util.stream.Stream;
+
+import static java.lang.Math.pow;
+import static java.lang.Math.sqrt;
+import static java.util.stream.Collectors.toList;
+
+/**
+ * Aggregate that finds standard deviation of a match query.
+ */
+class StdAggregate extends AbstractAggregate<Map<VarName, Concept>, Optional<Double>> {
+
+    private final VarName varName;
+
+    StdAggregate(VarName varName) {
+        this.varName = varName;
+    }
+
+    @Override
+    public Optional<Double> apply(Stream<? extends Map<VarName, Concept>> stream) {
+        List<Double> numbers = stream
+                .map(result -> result.get(varName).<Number>asResource().getValue().doubleValue())
+                .collect(toList());
+
+        OptionalDouble optAverage = numbers.stream().mapToDouble(x -> x).average();
+
+        if (!optAverage.isPresent()) {
+            return Optional.empty();
+        }
+
+        double average = optAverage.getAsDouble();
+
+        OptionalDouble variance = numbers.stream().mapToDouble(x -> pow(x - average, 2)).average();
+
+        if (!variance.isPresent()) {
+            return Optional.empty();
+        } else {
+            return Optional.of(sqrt(variance.getAsDouble()));
+        }
+    }
+
```

Comment:
Thanks, I've implemented the online version instead.

The results came out differently because that algorithm calculates the "unbiased sample" standard deviation. I'm unclear on the difference, so I've just clarified in the JavaDoc that this is the thing we're calculating.